generator/lookml.py

```diff
@@ -182,10 +183,22 @@ def _lookml(namespaces, glean_apps, target_dir, namespace_filter=[]):
     default=[],
     help="List of namespace names to generate lookml for.",
 )
-def lookml(namespaces, app_listings_uri, target_dir, metric_hub_repos, only):
+@click.option(
+    "--use_cloud_function",
+    "--use-cloud-function",
+    help="Use the Cloud Function to run dry runs during LookML generation.",
+    type=bool,
+)
+def lookml(
+    namespaces, app_listings_uri, target_dir, metric_hub_repos, only, use_cloud_function
+):
     """Generate lookml from namespaces."""
     if metric_hub_repos:
         MetricsConfigLoader.update_repos(metric_hub_repos)
 
     glean_apps = _get_glean_apps(app_listings_uri)
-    return _lookml(namespaces, glean_apps, target_dir, only)
+
+    dryrun = functools.partial(
+        DryRun, bigquery.Client(), use_cloud_function, id_token()
+    )
```

Comment:
Creating a partially applied `DryRun` instance and passing it turns out to be most efficient. 
Creating a `bigquery.Client` instance comes with some overhead, so doing it only once saves quite a bit of time (2x increase when creating a new client instance before each dryrun). Same goes for creating the ID token which is needed to invoke the dryrun cloud function (2x increase when creating the token before each dryrun)

Reply:
Should we do something similar in BigQuery ETL then?

Reply:
yeah, I'd be worth investigating whether we can make some improvements there